AudioDeviceExample/ViewController.swift

```diff
@@ -40,23 +44,81 @@ class ViewController: UIViewController {
         roomTextField.autocapitalizationType = .none
         roomTextField.delegate = self
-
-        prepareLocalMedia()
+        if #available(iOS 11.0, *) {
+            audioDeviceButton.isHidden = false;
+            logMessage(messageText: ViewController.coreAudioDeviceText + " Device selected")
+        } else {
+            audioDeviceButton.isHidden = true;
+        }
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
     }
 
+    @IBAction func selectAudioDevice() {
+        var coreAudioDeviceButton : UIAlertAction!
+        var engineAudioDeviceButton : UIAlertAction?
+
+        var selectedButton : UIAlertAction!
+
+        let alertController = UIAlertController(title: "Select Audio Device", message: nil, preferredStyle: .actionSheet)
+
+        // ExampleCoreAudioDevice
+        coreAudioDeviceButton = UIAlertAction(title: ViewController.coreAudioDeviceText,
+                                              style: .default,
+                                              handler: { (action) -> Void in
+            self.audioDevice = ExampleCoreAudioDevice()
+            self.logMessage(messageText: ViewController.coreAudioDeviceText + " Device Selected")
+        })
+        alertController.addAction(coreAudioDeviceButton!)
+
+        if #available(iOS 11.0, *) {
+            // EngineAudioDevice
+            engineAudioDeviceButton = UIAlertAction(title: ViewController.engineAudioDeviceText,
+                                                    style: .default,
+                                                    handler: { (action) -> Void in
+                self.audioDevice = ExampleEngineAudioDevice()
+                self.logMessage(messageText: ViewController.engineAudioDeviceText + " Device Selected")
+            })
+            alertController.addAction(engineAudioDeviceButton!)
+
+            if (self.audioDevice.isKind(of: ExampleCoreAudioDevice.self)) {
+                selectedButton = coreAudioDeviceButton
+            } else if (self.audioDevice.isKind(of: ExampleEngineAudioDevice.self)) {
+                selectedButton = engineAudioDeviceButton
+            }
+        } else {
+            selectedButton = coreAudioDeviceButton
+        }
+
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            alertController.popoverPresentationController?.sourceView = self.audioDeviceButton
+            alertController.popoverPresentationController?.sourceRect = self.audioDeviceButton.bounds
+        } else {
+            selectedButton?.setValue("true", forKey: "checked")
+
+            // Adding the cancel action
+            let cancelButton = UIAlertAction(title: "Cancel", style: .cancel, handler: { (action) -> Void in })
+            alertController.addAction(cancelButton)
+        }
+        self.navigationController!.present(alertController, animated: true, completion: nil)
+    }
```

Comment:
Should we add a `TVIDefaultAudioDevice` here?

Reply:
We decided not to for now. 👍